README.md

```diff
@@ -8,71 +8,117 @@
+We have well documented help commands so just run 
+
+```
+asyncapi --help 
 ```
-USAGE
+[back to top ⏫](#table-of-contents)
 
-asyncapi [options] [command]
 
-OPTIONS
 
--h, --help display help for command
--v, --version output the version number
+## API Reference
+
+### `validate`
 
-COMMANDS
+**USAGE**
 
-validate validate asyncapi file 
-context  Manage contexts
 ```
+asyncapi validate <spcPath | context-name> [options]
+```
+If you have current context set then 
+```
+asyncapi validate
+```
+
+**OPTIONS** 
 
-- #### `asyncapi validate --help`
 ```
-USAGE
+-h, --help Display help for command
+-w, --watch Enable watch mode (not implemented yet)
+```
 
-asyncapi validate [options]
+[back to top ⏫](#table-of-contents)
 
-OPTIONS
+## `context`
+
+Context is what makes it easier for you to work with multiple Asyncapi Files. You can add multiple different files to a context. This way you do not have to pass file as a input everytime you use any command. You can also set a default context, so neither you have to pass in filepath or context name as a input. 
+
+**USAGE**
 
--h, --help display help for command
--f, --file <spec-file-path> Path of the asyncapi file
--c, --context <saved-context-name>  context name to use
--w, --watch Enable Watch Mode (not implemented yet)
 ```
+asyncapi context [options] [command]
+```
+
+**COMMANDS**
+
+|command|arguments|description|example|
+|-------|---------|-----------|-------|
+|`list`|`none` |lists all saved context|`asyncapi context list`|
+|`current`|`none`|set current context|`asyncapi context current`|
+|`use` | `<context-name>`| set any context from the list as current|`asyncapi context use test`|
+|`add`|`<context-name> <spec-file-path>`|add/update a context|`asyncapi context add root ./asyncapi.yml`|
+|`remove`|`<context-name>`|remove a context from the list|`asyncapi context remove root`|
+
+**OPTIONS**
 
-- #### `asyncapi context --help`
 ```
-USAGE
+-h, --help display help for command
+```
 
-asyncapi context [options] [command] 
+[back to top ⏫](#table-of-contents)
 
-Context is what makes it easier for you to work with multiple AsyncAPI files.
-You can add multiple different files to a contextThis way you do not have to pass 
---file flag with path to the file every time but just --context flag with reference name
-You can also set a default context, so neither --file nor --context flags are needed.
 
-OPTIONS
+## Contributing
 
--h, --help display help for command 
+Read [CONTRIBUTING](https://github.com/asyncapi/CLI/blob/master/CONTRIBUTING.md) guide.
 
-COMMANDS
+### Setting up development enviornment
 
-list list of all saved contexts 
-current see current context 
-use <context-name> set given context as default/current
-add <context-name> <spec-file-path> add/update context
-remove <context-name> remove a context
+Follow these steps - 
+- Clone the repo.
+- Run `npm install` to install all the required dependencies
+- Run `npm run test` to make sure eveything is properly setup. 
```

Comment:
I think https://github.com/asyncapi/cli/issues/37 should be documented here

Reply:
Should I have a reference to this issue here or, have a brief paragraph talking about the convention we are following. And if we write about this wouldn't `CONTRIBUTING.MD` be the right place for this.

Reply:
it would be actually, you can add it to contributing guide

some inspiration where you have a link to large contrib guide + some extra -> https://github.com/kyma-project/console/blob/master/CONTRIBUTING.md

Reply:
@derberg what shall we call the section, does `Command Naming Pattern` sounds good.

Reply:
`Commands Structure and Patterns` sounds good I think?